device/cuda/src/finding/finding_algorithm.cu

```diff
@@ -204,80 +204,46 @@ finding_algorithm<stepper_t, navigator_t>::operator()(
-                                                                     m_mr.main);
-    bound_track_parameters_collection_types::device in_params(in_params_buffer);
-    bound_track_parameters_collection_types::const_device seeds(seeds_buffer);
-    thrust::copy(thrust::cuda::par.on(stream), seeds.begin(), seeds.end(),
-                 in_params.begin());
-
-    // Number of tracks per seed
-    vecmem::data::vector_buffer<unsigned int> n_tracks_per_seed_buffer(
-        n_seeds, m_mr.main);
-
-    // Create a map for links
-    std::map<unsigned int, vecmem::data::vector_buffer<candidate_link>>
-        link_map;
-
-    // Create a map for parameter ID to link ID
-    std::map<unsigned int, vecmem::data::vector_buffer<unsigned int>>
-        param_to_link_map;
-
-    // Create a map for tip links
-    std::map<unsigned int, vecmem::data::vector_buffer<
-                               typename candidate_link::link_index_type>>
-        tips_map;
-
-    // Link size
-    std::vector<std::size_t> n_candidates_per_step;
-    n_candidates_per_step.reserve(m_cfg.max_track_candidates_per_track);
-
-    std::vector<std::size_t> n_parameters_per_step;
-    n_parameters_per_step.reserve(m_cfg.max_track_candidates_per_track);
-
-    // Global counter object in Device memory
-    vecmem::unique_alloc_ptr<device::finding_global_counter>
-        global_counter_device =
-            vecmem::make_unique_alloc<device::finding_global_counter>(
-                m_mr.main);
-
-    // Global counter object in Host memory
-    device::finding_global_counter global_counter_host;
-
     /*****************************************************************
      * Measurement Operations
      *****************************************************************/
 
+    unsigned int n_modules;
     measurement_collection_types::const_view::size_type n_measurements =
         m_copy.get_size(measurements);
 
-    assert(is_contiguous_on<measurement_collection_types::const_device>(
-        measurement_module_projection(), m_mr.main, m_copy, m_stream,
-        measurements));
-
     // Get copy of barcode uniques
     measurement_collection_types::buffer uniques_buffer{n_measurements,
                                                         m_mr.main};
-    measurement_collection_types::device uniques(uniques_buffer);
 
-    measurement* end =
-        thrust::unique_copy(thrust::cuda::par.on(stream), measurements.ptr(),
-                            measurements.ptr() + n_measurements,
-                            uniques.begin(), measurement_equal_comp());
-    unsigned int n_modules = end - uniques.begin();
+    {
+        assert(is_contiguous_on<measurement_collection_types::const_device>(
+            measurement_module_projection(), m_mr.main, m_copy, m_stream,
+            measurements));
+
+        measurement_collection_types::device uniques(uniques_buffer);
+
+        measurement* end = thrust::unique_copy(
```

Comment:
I completely agree with the intention here. But I would go even one step further, as I have in #722, and call it `uniques_end` instead of `end`.

Since the scope is limited now, `end` is technically good enough. Still, as long as we're going for readability... 🤔